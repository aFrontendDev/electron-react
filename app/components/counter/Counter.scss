.backButton {
  position: relative;
}

.counter {
  font-size: 10rem;
  font-weight: bold;
}

.btnGroup {
  display: flex;
}

.btn {
  font-size: 1.6rem;
  font-weight: bold;
  background-color: #fff;
  border-radius: 50%;
  margin: 10px;
  width: 100px;
  height: 100px;
  opacity: 0.7;
  cursor: pointer;
}

.input {
  color: #000;

  &::placeholder {
    color: red;
  }
}

.btn:hover {
  color: white;
  background-color: rgba(0, 0, 0, 0.5);
}
