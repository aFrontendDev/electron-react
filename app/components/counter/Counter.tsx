import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Counter.scss';
import routes from '../../constants/routes.json';

type Props = {
  IncrementAction: () => void;
  IncrementIfOdd: () => void;
  IncrementAsyncAction: (delay?: number) => void;
  DecrementAction: () => void;
  count: number;
};

const Counter = (props: Props) => {
  const {
    IncrementAction,
    IncrementIfOdd,
    IncrementAsyncAction,
    DecrementAction,
    count
  } = props;

  return (
    <div>
      <input className={styles.input} type="text" placeholder="some text" />
      <div className={styles.backButton} data-tid="backButton">
        <Link to={routes.HOME}>BACK</Link>
      </div>
      <div className={`counter ${styles.counter}`} data-tid="counter">
        {count}
      </div>
      <div className={styles.btnGroup}>
        <button
          className={styles.btn}
          onClick={IncrementAction}
          data-tclass="btn"
          type="button"
        >
          +
        </button>
        <button
          className={styles.btn}
          onClick={DecrementAction}
          data-tclass="btn"
          type="button"
        >
          -
        </button>
        <button
          className={styles.btn}
          onClick={IncrementIfOdd}
          data-tclass="btn"
          type="button"
        >
          odd
        </button>
        <button
          className={styles.btn}
          onClick={() => IncrementAsyncAction()}
          data-tclass="btn"
          type="button"
        >
          async
        </button>
      </div>
    </div>
  );
};

export default Counter;
