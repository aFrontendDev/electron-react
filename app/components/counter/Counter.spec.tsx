/* eslint react/jsx-props-no-spreading: off */
import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Counter from './Counter';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const actions = {
    IncrementAction: spy(),
    IncrementIfOdd: spy(),
    IncrementAsyncAction: spy(),
    DecrementAction: spy()
  };
  const component = shallow(<Counter count={1} {...actions} />);
  return {
    component,
    actions,
    buttons: component.find('button'),
    p: component.find('.counter')
  };
}

describe('Counter component', () => {
  it('should should display count', () => {
    const { p } = setup();
    expect(p.text()).toMatch(/^1$/);
  });

  it('should first button should call increment', () => {
    const { buttons, actions } = setup();
    buttons.at(0).simulate('click');
    expect(actions.IncrementAction.called).toBe(true);
  });

  it('should match exact snapshot', () => {
    const { actions } = setup();
    const counter = (
      <div>
        <Router>
          <Counter count={1} {...actions} />
        </Router>
      </div>
    );
    const tree = renderer.create(counter).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should second button should call decrement', () => {
    const { buttons, actions } = setup();
    buttons.at(1).simulate('click');
    expect(actions.DecrementAction.called).toBe(true);
  });

  it('should third button should call IncrementIfOdd', () => {
    const { buttons, actions } = setup();
    buttons.at(2).simulate('click');
    expect(actions.IncrementIfOdd.called).toBe(true);
  });

  it('should fourth button should call incrementAsync', () => {
    const { buttons, actions } = setup();
    buttons.at(3).simulate('click');
    expect(actions.IncrementAsyncAction.called).toBe(true);
  });
});
