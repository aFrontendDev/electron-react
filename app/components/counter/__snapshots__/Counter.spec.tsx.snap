// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should match exact snapshot 1`] = `
<div>
  <div>
    <input
      className="input"
      placeholder="some text"
      type="text"
    />
    <div
      className="backButton"
      data-tid="backButton"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        BACK
      </a>
    </div>
    <div
      className="counter counter"
      data-tid="counter"
    >
      1
    </div>
    <div
      className="btnGroup"
    >
      <button
        className="btn"
        data-tclass="btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn"
        data-tclass="btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        data-tclass="btn"
        onClick={[Function]}
        type="button"
      >
        odd
      </button>
      <button
        className="btn"
        data-tclass="btn"
        onClick={[Function]}
        type="button"
      >
        async
      </button>
    </div>
  </div>
</div>
`;
